### PR TITLE
Infnoise: init at master

### PR DESCRIPTION
**Pull request by Stekke**
###### Motivation for this change

Init infnoise package, a driver for a USB random number generator.
At master because the latest release doesn't compile (problem finding the right headers).
The product: https://leetronics.de/en/shop/infinite-noise-trng/

Additionally update my maintainer info.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/drivers/infnoise/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchFromGitHub, git, libftdi }:
+
+stdenv.mkDerivation rec {
+  name = "infnoise";
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/infnoise/default.nix:4]:
```suggestion
  pname = "infnoise";
  version = "2019-08-12";
```

File: pkgs/misc/drivers/infnoise/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  src = fetchFromGitHub {
+    owner = "13-37-org";
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/infnoise/default.nix:7]: Why are we not using https://github.com/waywardgeek/infnoise ?

File: pkgs/misc/drivers/infnoise/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    repo = "${name}";
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/infnoise/default.nix:8]:
```suggestion
    repo = "infnoise";
```

File: pkgs/misc/drivers/infnoise/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    rev = "132683d4b5ce0902468b666cba63baea36e97f0c";
+    sha256 = "0r8kk8zwy4a4mwfwbq68hq23w88x5v2sf4zkyjizv9kb943mxs08";
+    leaveDotGit = true;
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/infnoise/default.nix:11]: Why is this required?

File: pkgs/misc/drivers/infnoise/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  nativeBuildInputs = [ git ];
+  buildInputs = [ libftdi ];
+
+  sourceRoot = "source/software";
+  makefile = "Makefile.linux";
+  makeFlags = [ "PREFIX=$(out)" ];
+  preInstall = ''
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/infnoise/default.nix:20]:
```suggestion
  postPatch = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/misc/drivers/infnoise/default.nix (hunk 2 above)
Comment (Stekke) [reply pkgs/misc/drivers/infnoise/default.nix:8]: I believe waywardgeek started this project, and  13-37-org commercialized it. Looking at releases and commits, it seems that 13-37-org is the current maintainer of the project. Debian is also using the 13-37-org repo (haven't checked other distro's).

File: pkgs/misc/drivers/infnoise/default.nix (hunk 4 above)
Comment (Stekke) [reply pkgs/misc/drivers/infnoise/default.nix:12]: During the build it gets info from the git repo to embed into the binary what commit it comes from, using define's.

File: pkgs/misc/drivers/infnoise/default.nix (hunk 5 above)
Comment (Stekke) [reply pkgs/misc/drivers/infnoise/default.nix]: Thanks for the review!

File: pkgs/misc/drivers/infnoise/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/misc/drivers/infnoise/default.nix:12]: Can we patch that to use version instead? Leaving dotGit is not fully reproducible.

Comment (Stekke) [reply pkgs/misc/drivers/infnoise/default.nix]: Ah, that's unfortunate...

File: pkgs/misc/drivers/infnoise/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{ lib, stdenv, fetchFromGitHub, libftdi }:
+
+stdenv.mkDerivation rec {
+  pname = "infnoise";
+  version = "2019-08-12";
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/infnoise/default.nix:5]:
```suggestion
  version = "unstable-2019-08-12";
```

Comment (Stekke) [reply pkgs/misc/drivers/infnoise/default.nix]: Makes sense

File: pkgs/misc/drivers/infnoise/default.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/13-37-org/infnoise";
+    description = "Driver for the Infinite Noise TRNG";
+    longDescription = ''
+      The Infinite Noise TRNG is a USB key hardware true random number generator.
+      It can either provide rng for userland applications, or provide rng for the OS entropy.
+      Add the following to your system configuration for plug and play support, adding to the OS entropy:
+      systemd.packages = [ pkgs.infnoise ];
+      services.udev.packages = [ pkgs.infnoise ];
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/infnoise/default.nix:37]: We could add a package module for this.

Comment (zhaofengli) [reply pkgs/misc/drivers/infnoise/default.nix:37]: Addressed in #171778.
